osu.Game.Rulesets.Taiko/UI/TaikoPlayfield.cs

```diff
@@ -146,6 +163,13 @@ private void load(OsuColour colours)
                 drumRollHitContainer.CreateProxy(),
             };
 
+            // to prioritise receiving key presses on input drum before objects, move input drum to the end of the hierarchy...
+            leftArea.Remove(inputDrum);
+            AddInternal(inputDrum);
+
+            // ...and create a proxy to keep the input drum displayed behind the playfield elements.
+            leftArea.Add(inputDrum.CreateProxy());
```

Comment:
Can you explain why this is done here and not inline in the above code? Is there some special meaning to adding it to one place then moving it to another?

Reply:
Initially had it such that the input drum will be initialized at the the end of the hierarchy, then proxied to the left area in subsequent logic. But simplified that to the following, which interestingly works:
```diff
diff --git a/osu.Game.Rulesets.Taiko/UI/TaikoPlayfield.cs b/osu.Game.Rulesets.Taiko/UI/TaikoPlayfield.cs
index 5891672f29..6d5c7812d5 100644
--- a/osu.Game.Rulesets.Taiko/UI/TaikoPlayfield.cs
+++ b/osu.Game.Rulesets.Taiko/UI/TaikoPlayfield.cs
@@ -65,6 +65,14 @@ private void load(OsuColour colours)
         {
             Container leftArea;

+            inputDrum = new InputDrum(HitObjectContainer)
+            {
+                Anchor = Anchor.CentreLeft,
+                Origin = Anchor.CentreLeft,
+                AutoSizeAxes = Axes.X,
+                RelativeSizeAxes = Axes.Y,
+            };
+
             InternalChildren = new[]
             {
                 new SkinnableDrawable(new TaikoSkinComponent(TaikoSkinComponents.PlayfieldBackgroundRight), _ => new PlayfieldBackgroundRight()),
@@ -74,16 +82,10 @@ private void load(OsuColour colours)
                     RelativeSizeAxes = Axes.Both,
                     FillMode = FillMode.Fit,
                     BorderColour = colours.Gray0,
-                    Children = new Drawable[]
+                    Children = new[]
                     {
                         new SkinnableDrawable(new TaikoSkinComponent(TaikoSkinComponents.PlayfieldBackgroundLeft), _ => new PlayfieldBackgroundLeft()),
-                        inputDrum = new InputDrum(HitObjectContainer)
-                        {
-                            Anchor = Anchor.CentreLeft,
-                            Origin = Anchor.CentreLeft,
-                            AutoSizeAxes = Axes.X,
-                            RelativeSizeAxes = Axes.Y,
-                        },
+                        inputDrum.CreateProxy(),
                     }
                 },
                 mascot = new SkinnableDrawable(new TaikoSkinComponent(TaikoSkinComponents.Mascot), _ => Empty())
@@ -161,15 +163,9 @@ private void load(OsuColour colours)
                     RelativeSizeAxes = Axes.Both,
                 },
                 drumRollHitContainer.CreateProxy(),
+                inputDrum,
             };

-            // to prioritise receiving key presses on input drum before objects, move input drum to the end of the hierarchy...
-            leftArea.Remove(inputDrum);
-            AddInternal(inputDrum);
-
-            // ...and create a proxy to keep the input drum displayed behind the playfield elements.
-            leftArea.Add(inputDrum.CreateProxy());
-
             RegisterPool<Hit, DrawableHit>(50);
             RegisterPool<Hit.StrongNestedHit, DrawableHit.StrongNestedHit>(50);
```

Reply:
I'd keep the comment explaining the non-proxy placement alongside it, but seems fine.